src/coreComponents/constitutive/fluid/multifluid/MultiFluidBase.hpp

```diff
@@ -783,12 +783,14 @@ MultiFluidBase::KernelWrapper::
   real64 totalMolality = 0.0;
   for( integer ic = 0; ic < numComps; ++ic )
   {
+    GEOS_ERROR_IF( m_componentMolarWeight[ic] < LvArray::NumericLimits< real64 >::epsilon, "Zero component molecular weight" );
```

Comment:
`m_componentMolarWeight` is user input so we should validate it in `postInputInitialization` [MultiFluidBase.cpp:184](https://github.com/GEOS-DEV/GEOS/blob/develop/src/coreComponents/constitutive/fluid/multifluid/MultiFluidBase.cpp#L184) instead of in this kernel function.

Reply:
moved the check there, thanks